_posts/en/newsletters/2021-11-10-newsletter.md

```diff
@@ -0,0 +1,225 @@
+      discrepancies between peers.
+
+    - *Multi-block filters:* these could further reduce the data
+      lightweight clients will need to download.
+
+    - *Committed block filters:* requiring miners commit to the filter
+      for their blocks, reducing the amount of data lightweight clients
+      need to download to monitor for discrepancies between the filters
+      being served by different peers.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review Club][]
+meeting, highlighting some of the important questions and answers.  Click on a
+question below to see a summary of the answer from the meeting.*
+
+[Add `ChainstateManager::ProcessTransaction`][review club #23173] is a PR by
+John Newbery to add a new `ChainstateManager::ProcessTransaction()` interface
+function responsible for processing transactions as candidates to the mempool
+and performing mempool consistency checks. The review club discussed the
+current interface for adding transactions to the mempool.
+
+{% include functions/details-list.md
+  q0="What is `cs_main`? Why is it called `cs_main`?"
+  a0="`cs_main` is a mutex intended to synchronize multi-threaded access to
+validation state. In reality, it also guards non-validation data in
+net\_processing, and multiple contributors wish to minimize the usage of`cs_main`.
+The variable was named when validation functionality was housed in a main.cpp
+file. The prefix, `cs`, stands for critical section."
+a0link="https://bitcoincore.reviews/23173#l-45"
+
+  q1="Which components currently call `AcceptToMemoryPool`? Which of the ATMP calls are from external client code and which are from inside validation?"
+  a1="Excluding calls from tests, there are four call sites:
+  1. When the node starts, it [loads][atmp disk] transactions from mempool.dat and
+     calls ATMP to re-validate the transactions and restore mempool contents. This
+     call originates from a component 'external' to validation.
+  2. Transactions received from peers on the P2P network are [validated and submitted
+   to the mempool][atmp p2p] through ATMP.
+  3. During a reorg, any transactions that were present in the disconnected
+     block(s) but not included in the new chain tip are [resubmitted][atmp reorg]
+     to the mempool using ATMP. This is an internal validation call.
+  4. Clients such as RPC (e.g. `sendrawtransaction`) and the wallet (e.g.
+     `sendtoaddress`) submit their transactions to the node using
+     [`BroadcastTransaction()`][atmp client], which calls ATMP. The
+     `testmempoolaccept` RPC also calls ATMP with `test_accept` set to `true`. These
+     are examples of calls from components 'external' to validation."
+  a1link="https://bitcoincore.reviews/23173#l-80"
+
+  q2="What does `CTxMemPool::check()` do? Whose responsibility is it to call
+    that function?"
+  a2="`CTxMemPool::check()` checks that each transaction's inputs correspond to
```

Comment:
I don't think this makes it clear that that this is checking the entire contents of the mempool (and could instead be checking transactions that are being submitted to the mempool). How about something like:

```suggestion
  a2="`CTxMemPool::check()` does checks on _all_ the transactions currently in the mempool. It checks that each transaction's inputs correspond to
```

Reply:
thanks, changed to "checks that all transactions' inputs correspond to available UTXOs and performs internal consistency checks over the entire mempool."